src/mongocxx/test/spec/util.cpp

```diff
@@ -665,134 +835,161 @@ void run_transaction_operations(document::view test,
+
+                    // Save their lsids so they are available after calling `endSession`.
+                    session_lsid0.reset(session0.id());
+                    session_lsid1.reset(session1.id());
+
+                    // The test runner MUST also ensure that the configureFailPoint command is
+                    // excluded from the list of observed command monitoring events for this client
+                    // (if applicable).
+                    apm_checker.clear();
+                    apm_checker.set_ignore_command_monitoring_event("configureFailPoint");
+
+                    // If a test uses `targetedFailPoint`, disable the fail point after running all
+                    // `operations` to avoid spurious failures in subsequent tests.
+                    stdx::optional<targeted_fail_point_guard_type> targeted_fail_point_guard;
+
+                    // Step 11. Perform the operations.
+                    run_transaction_operations(test.get_document().value,
+                                               &client,
+                                               db_name,
+                                               coll_name,
+                                               &session0,
+                                               &session1,
+                                               &targeted_fail_point_guard,
+                                               apm_checker);
+
+                    // Step 12: Call session0.endSession() and session1.endSession (via
+                    // client_session dtors).
+                }
+
+                // Step 13. If the test includes a list of command-started events in expectations,
+                // compare them to the actual command-started events using the same logic as the
+                // legacy Command Monitoring Spec Tests runner, plus the rules in the
+                // Command-Started Events instructions.
+                test_util::match_visitor visitor =
+                    [&](bsoncxx::stdx::string_view key,
+                        bsoncxx::stdx::optional<bsoncxx::types::bson_value::view> main,
+                        bsoncxx::types::bson_value::view pattern) {
+                        if (key.compare("lsid") == 0) {
+                            REQUIRE(pattern.type() == type::k_string);
+                            REQUIRE(main);
+                            REQUIRE(main->type() == type::k_document);
+                            auto session_name = pattern.get_string().value;
+                            if (session_name.compare("session0") == 0) {
+                                REQUIRE(
+                                    test_util::matches(session_lsid0, main->get_document().value));
+                            } else {
+                                REQUIRE(
+                                    test_util::matches(session_lsid1, main->get_document().value));
+                            }
```

Comment:
```suggestion
                            }
                            return test_util::match_action::k_skip;
```

Skip when matching `lsid`. The pattern value is a string "session0" / "session1". Actual value is the "lsid" document.